theano/scan_module/scan_op.py

```diff
@@ -402,6 +402,32 @@ def make_node(self, *inputs):
                     'by using dimshuffle or shape_padleft. '
                     )
 
+        def _check_broadcast(v1, v2):
+            """ Checks that the broadcast pattern of v1 and v2.
+
+            Controls that the broadcast pattern of the variable provided as
+            input to `scan` matches the broadcast pattern provided in
+            `output_info`. It raises an error when they don't match. The
+            typical case is when the user provides either the input or the
+            `output_info` (but not both) with a dimension fixed to 1,
+            which may wrongly be interpreted as broadcastable.
```

Comment:
The method format() is call for other thing then output_info. I think this method _check_broadcast should not be called inside format(), but before the format() call for the handling of output info.